File: frontend/src/api/execution.ts
import client from './client'

export interface SessionStatus {
  session_id: string
  pipeline_id: string
  status: 'running' | 'stopped' | 'error' | 'completed'
  started_at: string
  stopped_at?: string
  error_msg?: string
}

export const executionApi = {
  start: (pipelineId: string, paramsOverride?: Record<string, unknown>): Promise<{ session_id: string }> =>
    client.post('/execution/start', { pipeline_id: pipelineId, params_override: paramsOverride }).then((r) => r.data),

  stop: (sessionId: string): Promise<void> =>
    client.post(`/execution/stop/${sessionId}`).then(() => undefined),

  status: (sessionId: string): Promise<SessionStatus> =>
    client.get(`/execution/status/${sessionId}`).then((r) => r.data),

  logs: (sessionId: string): Promise<string[]> =>
    client.get(`/execution/logs/${sessionId}`).then((r) => r.data),

  sessions: (): Promise<SessionStatus[]> =>
    client.get('/execution/sessions').then((r) => r.data),
}

File: backend/app/schemas/execution.py
from datetime import datetime
from typing import Any
from pydantic import BaseModel


class ExecutionStartRequest(BaseModel):
    pipeline_id: str
    params_override: dict[str, Any] | None = None
    mode: str = "sequential"  # "sequential" | "multiprocess"


class ExecutionStartResponse(BaseModel):
    session_id: str
    ws_port: int = 8765


class SessionStatusResponse(BaseModel):
    session_id: str
    pipeline_id: str
    status: str        # running | stopped | error | completed
    started_at: datetime
    stopped_at: datetime | None = None
    error_msg: str | None = None
